packages/dashmate/src/commands/config/set.js

```diff
@@ -20,10 +22,36 @@ class ConfigSetCommand extends ConfigBaseCommand {
       optionValue = null;
     }
 
-    config.set(optionPath, optionValue);
+    // check for existence
+    config.get(optionPath);
+
+    const path = optionPath.split('.');
+
+    let schema = await $RefParser.dereference(configJsonSchema);
+
+    for (const e of path) {
+      if (schema.$ref) {
```

Comment:
You don't need to handle ref. You can use this function https://github.com/dashpay/platform/blob/32433e2b34694e029f394fb1baa9360e2d9bec1b/packages/js-dpp/lib/dataContract/getPropertyDefinitionByPath.js